src/super_gradients/training/metrics/detection_metrics.py

```diff
@@ -89,19 +91,30 @@ def ap_per_class(tp, conf, pred_cls, target_cls):
 
 
 class DetectionMetrics(Metric):
-    def __init__(self, num_cls,
-                 post_prediction_callback: DetectionPostPredictionCallback = None,
-                 iou_thres: IouThreshold = IouThreshold.MAP_05_TO_095,
-                 dist_sync_on_step=False):
-        """
+    """
+    DetectionMetrics
```

Comment:
don't you think it will make more sense to define normalize_targets in the callback instead?

Metrics are not expected to change the data, but a post prediction callback is